### ISSUE TITLE
File/Close option throws exception

### ISSUE THREAD

**Issue by rat-moonshine** [prominic/Moonshine-IDE#102]
While selecting option from File/Close menu, Moonshine throws following exception:

![image](https://user-images.githubusercontent.com/24370870/33755499-08da5d10-dc17-11e7-8d3f-cd216256dc28.png)

**Comment by JoelProminic** [comment]
Here is the behavior I see with File > Close:

1. Open a project
2. Open a file from the project
3. File > Close
4. The tab closes properly
5. Close all open tabs
6. File > Close
7. BUG:  A "New" tab opens
8. File > Close
9. BUG:  Nothing happens

**Comment by JoelProminic** [comment]
The File > Close bug does not trigger for me now.

**Comment by rat-moonshine** [comment]
It's becoming hard to find out where an `Event` type casting as `CloseTabEvent`. as the usage were numerous across multiple projects. Fixing in earlier method (where I able to pin-point it) didn't resolve this problem completely. 

I noticed that `CloseTabEvent.EVENT_CLOSE_TAB` fires in dual ways:
- When dispatched through some classes conditionally as `CloseTabEvent.EVENT_CLOSE_TAB`
- When dispatches by clicking on top menubar option which dispatches as `Event`

Thus, its possible that at somewhere any listener method type casting becoming wrong:
```
private function onCloseTabEventFired(event:CloseTabEvent):void
```
All such handler event type now changed as following, and further checks added:
```
private function onCloseTabEventFired(event:Event):void
{
      if (event is CloseTabEvent)
      {
            var tmpEvent:CloseTabEvent = event as CloseTabEvent;
            // do further process with tmpEvent's (CloseTabEvent) properties
```

While closing Moonshine, a save alert was populating asking to save an unsaved editor - when there was no editor opened at a time. This is now fixed.
